compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp

```diff
@@ -0,0 +1,172 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  =============================================================================*/
+
+#include "ConfigurationManager.hpp"
+
+namespace cppmicroservices {
+namespace scrimpl {
+ConfigurationManager::ConfigurationManager(
+  const std::shared_ptr<const metadata::ComponentMetadata> metadata,
+  const cppmicroservices::BundleContext& bc,
+  std::shared_ptr<cppmicroservices::logservice::LogService> logger)
+  : metadata(metadata)
+  , logger(std::move(logger))
+  , bundleContext(bc)
+  , mergedProperties (metadata->properties)
+{
+  if (!this->metadata || !this->bundleContext || !this->logger) {
+    throw std::invalid_argument(
+      "ConfigurationManagerImpl - Invalid arguments passed to constructor");
+  }
+}
+
+cppmicroservices::AnyMap ConfigurationManager::GetProperties() const noexcept
+{
+  std::lock_guard<std::mutex> lock(propertiesMutex);
+
+  return mergedProperties;
+}
+
+void ConfigurationManager::Initialize() {
+  if ((metadata->configurationPids.empty()) ||
+      (metadata->configurationPolicy == metadata::ComponentMetadata::CONFIG_POLICY_IGNORE))
+    return;
```

Comment:
always use brackets even for one line if statements.

Reply:
Fixed